src/compiler/tsc.ts

```diff
@@ -146,131 +173,176 @@ module ts {
             return sys.exit(EmitReturnStatus.CompilerOptionsErrors);
         }
 
-        if (compilerOptions.version) {
+        if (commandLine.options.version) {
             reportDiagnostic(createCompilerDiagnostic(Diagnostics.Version_0, version));
             return sys.exit(EmitReturnStatus.Succeeded);
         }
 
-        if (compilerOptions.help) {
+        if (commandLine.options.help) {
             printVersion();
             printHelp();
             return sys.exit(EmitReturnStatus.Succeeded);
         }
 
-        if (commandLine.filenames.length === 0) {
+        if (commandLine.options.project) {
+            configFilename = normalizePath(combinePaths(commandLine.options.project, "tsconfig.json"));

```

Comment:
JSON is not supported on scscript. this will crash then. i would add an error message The_current_host_does_not_support_the_0_option similar to locale and watch.